### ISSUE TITLE
Breaking changes in migration runner between 3.3.2 and 5.0.0

### ISSUE THREAD

**Issue by Ross-Gargett** [fluentmigrator/fluentmigrator#1732]
**Describe the bug**
The in-process runner from version 5.0.0 is not backwards compatible with migrations written with 3.3.2. We upgraded our migration tool and some of our projects to .NET 8 and FluentMigrator 5.0.0. The migration runner tool throws an exception for some migrations that are using 3.3.2. The one I encountered was around `Delete.Table()` which seems to have breaking interface changes, see the stack trace below. 

```
System.MissingMethodException: Method not found: 'FluentMigrator.Builders.IInSchemaSyntax FluentMigrator.Builders.Delete.IDeleteExpressionRoot.Table(System.String)'.
   at ...MigrationName...
   at FluentMigrator.MigrationBase.GetUpExpressions(IMigrationContext context)
   at FluentMigrator.Runner.MigrationRunner.<>c.<ApplyMigrationUp>b__70_0(IMigration m, IMigrationContext c)
   at FluentMigrator.Runner.MigrationRunner.ExecuteMigration(IMigration migration, Action`2 getExpressions)
   at FluentMigrator.Runner.MigrationRunner.ApplyMigrationUp(IMigrationInfo migrationInfo, Boolean useTransaction)
   at FluentMigrator.Runner.MigrationRunner.MigrateUp(Int64 targetVersion, Boolean useAutomaticTransactionManagement)
   at FluentMigrator.Runner.MigrationRunner.MigrateUp(Boolean useAutomaticTransactionManagement)
   at FluentMigrator.Runner.MigrationRunner.MigrateUp()
...
```

**To Reproduce**
Use the FluentMigrator 5.0.0 in-process migration runner to migrate up a project which deletes a table using FluentMigrator 3.3.2.

**Expected behavior**
Migration runner should be backwards compatible if there are no documented breaking changes.

**Information (please complete the following information):**
 - OS: MacOS 14.2.1
 - Platform: .NET 6 and .NET 8
 - FluentMigrator version: 3.3.2 -> 5.0.0
 - FluentMigrator runner: "in-process runner" 
 - Database Management System:  SQLServer
 - Database Management System Version: Rider

**Additional context**
We have a tool we maintain which we use to run database migrations for multiple projects.
We are upgrading some of these projects, and this tool to .NET 8. Since this tool is used across many different projects, it needs to be backwards compatible with older versions (until there are breaking changes).

**Comment by jzabroski** [comment]
The breaking change you are referring to is this one: https://github.com/fluentmigrator/fluentmigrator/commit/b758e9cbbc8d2e625cf470d7d604ce57236b4e30

In general, we can't provide a backwards compatible interface for this, since C# does not support return type covariance.

I disagree that it **needs** to be backward compatible. There are very few projects out there that still support .NET Framework Classic applications.  I get why you want me to support it - it's probably less work for you.

As an alternative, why cannot your team use 3.3.2 for old projects not yet upgraded, and .NET 8 for new projects, and run .NET 8 migrations using FluentMigrator 5.0, and .NET Framework Classic migrations using 3.3.2? I am guessing this is not a lot of work for your team to support.

**Comment by jzabroski** [comment]
I have fixed the 5.0.0 release notes to retroactively flag this breaking change. 

https://github.com/fluentmigrator/fluentmigrator/releases/tag/v5.0.0

Thanks for reporting it!

**Comment by Ross-Gargett** [comment]
Thanks for the follow-up and documentation changes.
